core/templates/pages/contributor-dashboard-page/services/contribution-and-review.service.spec.ts

```diff
@@ -325,6 +365,252 @@ describe('Contribution and review service', () => {
+              default_outcome: {}
+            }
+          },
+          interaction: {
+            answer_groups: [],
+            confirmed_unclassified_answers: [],
+            customization_args: {
+              recommendedExplorationIds: {
+                value: []
+              }
+            },
+            default_outcome: null,
+            hints: [],
+            solution: null,
+            id: 'EndExploration'
+          },
+          linked_skill_id: null,
+          param_changes: [],
+          solicit_answer_details: false,
+          card_is_checkpoint: false
+        }
+      };
+      const mockReadOnlyExplorationData: FetchExplorationBackendResponse = {
+        can_edit: true,
+        exploration: {
+          init_state_name: 'Introduction',
+          param_changes: [],
+          param_specs: {},
+          states: mockStates,
+          title: 'Dummy Title',
+          language_code: 'en',
+          objective: 'Dummy Objective',
+          next_content_id_index: 4,
+          correctness_feedback_enabled: true
+        },
+        exploration_metadata: {
+          title: 'Dummy Title',
+          category: 'Dummy Category',
+          objective: 'Dummy Objective',
+          language_code: 'en',
+          tags: [],
+          blurb: 'Dummy Blurb',
+          author_notes: 'Dummy Notes',
+          states_schema_version: 0,
+          init_state_name: 'Introduction',
+          param_specs: {},
+          param_changes: [],
+          auto_tts_enabled: true,
+          correctness_feedback_enabled: true,
+          edits_allowed: true,
+        },
+        exploration_id: '1',
+        is_logged_in: true,
+        session_id: '0',
+        version: 0,
+        preferred_audio_language_code: 'en',
+        preferred_language_codes: [],
+        auto_tts_enabled: true,
+        correctness_feedback_enabled: true,
+        record_playthrough_probability: 1,
+        draft_change_list_id: 1,
+        has_viewed_lesson_info_modal_once: false,
+        furthest_reached_checkpoint_exp_version: 0,
+        furthest_reached_checkpoint_state_name: '',
+        most_recently_reached_checkpoint_state_name: '',
+        most_recently_reached_checkpoint_exp_version: 1,
+        displayable_language_codes: ['en'],
+      };
+      const explorationBackendDict = explorationObjectFactory.
+      createBackendDictFromExplorationBackendResponse(mockReadOnlyExplorationData);
+      const exploration = explorationObjectFactory.createFromBackendDict(
+        explorationBackendDict);
+      // const getStatesSpy = spyOn(exploration, 'getStates');
+
+      fetchExplorationSpy.and.returnValue(
+        Promise.resolve(mockReadOnlyExplorationData));
+      fetchSuggestionsAsyncSpy.and.returnValue(
+        Promise.resolve(backendFetchResponse));
+      explorationObjectFactorySpy.and.returnValue(
+        Promise.resolve(exploration));
+      // getStatesSpy.and.returnValue(
+      //   Promise.resolve(mockStates));
```

Comment:
Remove? Please make sure to do do your own pass to catch things like this.

Reply:
Done